src/elements/content-explorer/ContentExplorer.js

```diff
@@ -751,21 +751,20 @@ class ContentExplorer extends Component<Props, State> {
      *
      * @private
      * @param {Collection} collection - collection that needs to be updated
-     * @param {?BoxItem} selected - item that should be selected in that collection (if present)
+     * @param {string} [selectedId] - ID of the item that should be selected in that collection (if present)
      * @param {Function} [callback] - callback function that should be called after setState occurs
      * @return {void}
      */
-    updateCollection(collection: Collection, selected: ?BoxItem, callback: Function = noop): Object {
+    updateCollection(collection: Collection, selectedId: ?string, callback: Function = noop): Object {
```

Comment:
Changed this method to just take in the id of the selected item, to remove the confusion about which is the source of truth